images/csp_violation_report_service/app/tests/unit/controllers/test_ReportController.py

```diff
@@ -0,0 +1,151 @@
+import json
+from tests import TestCase
+from app.controllers.ReportController import ReportController
+
+from unittest.mock import ANY, MagicMock, patch
+
+
+class ReportControllerTest(TestCase):
+    @patch("app.controllers.ReportController.Report")
+    @patch("app.controllers.ReportController.Domain")
+    def test_save(self, mock_domain, mock_report):
+        controller = ReportController()
+        request = MagicMock()
+        request.input.return_value = {
+            "document-uri": "https://example.com/foo/bar",
+            "referrer": "https://www.google.com/",
+            "violated-directive": "default-src self",
+            "original-policy": "default-src self; report-uri /csp-hotline.php",
+            "blocked-uri": "http://evilhackerscripts.com",
```

Comment:
I mean the fact that this is http only is just a sad comment on the state of evil hackers in general.